public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx

```diff
@@ -236,21 +243,38 @@ export const VisualizationContainer = ({
   }
 
   const fetchVisualization = async () => {
-    return savedVisualizationId
-      ? await fetchVisualizationById(http, savedVisualizationId, setIsError)
-      : inputMetaData;
+    try {
+      return savedVisualizationId
+        ? await fetchVisualizationById(http, savedVisualizationId, setIsError)
+        : inputMetaData;
+    } catch (error) {
+      setIsError({ error });
+      return null;
+    }
```

Comment:
What are the cases where visualization will fail and reach the catch error?

Reply:
was trying to fix a bug and modified this, but its not needed so reverting it back